hack/chains/end-to-end-demo.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+#
+# A demo script that runs a build pipeline including HACBS specific tests and a sample release pipeline with HACBS Enterprise Contract test
+
+APP_GIT_REPOSITORY=https://github.com/jduimovich/single-nodejs-app
+APP_BUILDER=nodejs-builder
+APP_IMAGE_NAME="${APP_GIT_REPOSITORY/*\/}"
+APP_IMAGE_REF="quay.io/${MY_QUAY_USER}/${APP_IMAGE_NAME}"
+
+# Check for $MY_QUAY_USER
+if [[ -z ${MY_QUAY_USER+x} ]]; then
+  echo "🛑 Make sure to set the \$MY_QUAY_USER environment variable, it needs to be set to the
+   Quay.io organization where the resulting images will be pushed"
+  exit 1
+fi
+
+set -euo pipefail
+
+echo "ℹ️  Please wait..."
+
+DOCKER_LOGIN=$(docker login quay.io </dev/null 2>&1 || true)
+if [[ "${DOCKER_LOGIN}" != *"Login Succeeded"* ]]; then
+  echo "ℹ️  Please enter the quay.io credentials for ${MY_QUAY_USER}, if you're using a robot account
+and token make sure that it's allowed to push to ${APP_IMAGE_NAME} repository.
+"
+  docker login quay.io --username "${MY_QUAY_USER}"
+fi
+
+# Making the quay.io repository public, this is a requirement right now for the Test stream tasks to succeed
+if [[ "$(curl -s "https://quay.io/api/v1/repository?namespace=${MY_QUAY_USER}&public=true" | jq ".repositories[] | select(.name==\"${APP_IMAGE_NAME}\") | has(\"name\")")" != 'true' ]]; then
+  echo "🛑 Make sure that the quay.io repository '${APP_IMAGE_NAME}' exist and it's visibility
+is set to public, create an empty repository here:
+
+👉 https://quay.io/new/?name=${APP_IMAGE_NAME}
+
+if the repository already exists change the visibility here:
+
+👉 https://quay.io/repository/${MY_QUAY_USER}/${APP_IMAGE_NAME}?tab=settings "
+  exit 1
+fi
+
+# We're going to assume running with kubeadmin
+if [[ "$(kubectl auth can-i '*' '*' < /dev/null 2>&1)" != 'yes' ]]; then
+  echo "🛑 Make sure that you're logged in as kubeadmin, we make assumptions here"
+  exit 1
+fi
+
+# Checking the status of the build component as a way to make sure that the cluster is okay to run the demo
+if [[ "$(kubectl get -n openshift-gitops applications.argoproj.io build -o jsonpath='{.status.sync.status}')" != 'Synced' ]]; then
+  echo "🛑 Make sure that the cluster is setup by running hack/bootstrap-cluster.sh first see
```

Comment:
Maybe a more literal message here is useful, e.g. "The build application is not synced. Make sure ..."